lib/data/multipleBackendGateway.js

```diff
@@ -0,0 +1,68 @@
+import { errors } from 'arsenal';
+import { Logger } from 'werelogs';
+
+import config from '../Config';
+import parseLC from './locationConstraintParser';
+
+const logger = new Logger('MultipleBackendGateway', {
+    logLevel: config.log.logLevel,
+    dumpLevel: config.log.dumpLevel,
+});
+
+function createLogger(reqUids) {
+    return reqUids ?
+        logger.newRequestLoggerFromSerializedUids(reqUids) :
+        logger.newRequestLogger();
+}
+
+const clients = parseLC(config);
+
+const multipleBackendGateway = {
+    put: (stream, size, keyContext, backendInfo, reqUids, callback) => {
+        const controllingLocationConstraint =
+            backendInfo.getControllingLocationConstraint();
+        const client = clients[controllingLocationConstraint];
+        if (!client) {
+            const log = createLogger(reqUids);
+            log.error('no data backend matching controlling locationConstraint',
+            { controllingLocationConstraint });
+            return process.nextTick(() => {
+                callback(errors.InternalError);
+            });
+        }
+        return client.put(stream, size, keyContext,
+            reqUids, (err, key) => {
+                if (err) {
+                    const log = createLogger(reqUids);
+                    log.error('error from datastore',
+                             { error: err, implName: client.clientType });
+                    return callback(errors.InternalError);
+                }
+                const dataRetrievalInfo = {
+                    key,
+                    dataStoreName: controllingLocationConstraint,
+                };
+                return callback(null, dataRetrievalInfo);
+            });
+    },
+
+    get: (objectGetInfo, range, reqUids, callback) => {
+        const client = clients[objectGetInfo.dataStoreName];
+        if (client.clientType === 'scality') {
```

Comment:
yes, we could add that to sproxyd client.  metadata uses sproxyd client as well for backups so we were trying to not mess with it but that change would not break anything.